### PR TITLE
Add support for GCS reserve memory

### PR DESCRIPTION
**Pull request by jterry75**
Signed-off-by: Justin Terry (VM) <juterry@microsoft.com>

### CODE REVIEW

**Comment by kevpar** [comment]
Should we default the gcs limit to 0 (no limit) instead? I'm not sure what the limit accomplishes since if it's hit, the whole pod will go down anyways, and is currently hard to troubleshoot in that state. We could leave in the ability to configure a limit via command line still.

File: service/gcs/main.go [hunk 1]
```diff
@@ -126,7 +131,7 @@ func main() {
 	if err := syscall.Sysinfo(&sinfo); err != nil {
```
Comment (mkatri) [inline service/gcs/main.go]: should the 75MB number in the comment above be increased to 110MB?

**Comment by mkatri** [comment]
> Should we default the gcs limit to 0 (no limit) instead? I'm not sure what the limit accomplishes since if it's hit, the whole pod will go down anyways, and is currently hard to troubleshoot in that state. We could leave in the ability to configure a limit via command line still.

Wouldn't we at least know which container misbehaved? GCS or workload?

File: service/gcs/main.go (hunk 1 above)
Comment (jterry75) [reply service/gcs/main.go]: Good catch the comment is now wrong.

**Comment by jterry75** [comment]
> Should we default the gcs limit to 0 (no limit) instead? I'm not sure what the limit accomplishes since if it's hit, the whole pod will go down anyways, and is currently hard to troubleshoot in that state. We could leave in the ability to configure a limit via command line still.

I think as a best practice we should keep the gcs cgroup and use its limit. I agree that it will crash if it hits this limit but it would be nice to know. There is no reason that we should be seeing 50mb of usage and if we do keep seeing crashes we know this is not the real issue. Without the limit there is no way to know if a gcs is running with 200MB and stealing from workload memory.

**Comment by jterry75** [comment]
**DO NOT MERGE** doing some studies to try and find the right numbers here.

**Comment by kevpar** [comment]
Closing this since we took another fix for this issue: https://github.com/microsoft/opengcs/pull/372

**Pull request closed without merging**